packages/backend/src/controllers/membership-plan.ts

```diff
@@ -1,18 +1,107 @@
 import { RESTController } from '../libs/classes/RESTController';
 import { getRepository } from 'typeorm';
-import { Request, Response } from 'express';
+import { NextFunction, Request, Response } from 'express';
 import { MembershipPlan, PlanType } from '../models/MembershipPlan';
+import { User } from '../models/User';
 
 export class MembershipPlanCtrl extends RESTController<MembershipPlan> {
   constructor() {
     super(getRepository(MembershipPlan));
   }
 
+  /**
+   * 
+   * @param req 
+   * @param res 
+   */
   public getAllTypes = async (
     req: Request,
     res: Response
   ): Promise<Response> => {
-    return res.status(201).send(Object.values(PlanType)
-    .filter(type => typeof type === 'string')); //Return the list of type's names. We probably have to do a refactoring about membershipplan's types like have a specific table in the db.
+    return res
+      .status(201)
+      .send(Object.values(PlanType).filter((type) => typeof type === 'string')); //Return the list of type's names. We probably have to do a refactoring about membershipplan's types like have a specific table in the db.
+  };
+
+  /**
+   * 
+   * @param req 
+   * @param res 
+   */
+  public getAll = async (req: Request, res: Response): Promise<Response> => {
+    if (req.user.user.group === 'admin') {
+      return res.send(await this.findAll());
+    }
+
+    const userRepo = getRepository(User);
+    const currentUser = await userRepo.findOne(req.user.user.id);
+    const currentOrg = await currentUser.getUserOrganisation();
+
+    return res.send(
+      await this.findAll({
+        where: { organisation: currentOrg.id }
+      })
+    );
+  };
+
+  /**
+   * 
+   * @param req 
+   * @param res 
+   */
+  public getOne = async (req: Request, res: Response): Promise<Response> => {
+    const id = Number.parseInt(req.params.id);
+
+    if (req.user.user.group === 'admin') {
+      return res.send(await this.repository.findOneOrFail(id, {}));
```

Comment:
You said, a long time ago, it is better to first set a variable :

const membershipPlan = await this.repository.findOneOrFail(id, {})
return res.send(membershipPlan )